### PR TITLE
docs update for table primitive, use theme values, use exampleCode co…

### PR DESCRIPTION
**Pull request by jacoblogan**
*Issue #944

*Description of changes: Update to the docs Table component, update examples, use theme values.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/src/pages/components/table/examples/index.ts [hunk 1]
```diff
@@ -1,5 +1,8 @@
+import { TableFontSizePropExample } from './TableFontSizePropExample';
 export { BasicExample } from './basicExample';
```
Comment (zchenwei) [inline docs/src/pages/components/table/examples/index.ts:2]: Could we update the old file names to be Pascal case?

File: docs/src/pages/components/table/TablePropControls.tsx [hunk 2]
```diff
@@ -7,8 +7,8 @@ import {
   TextField,
 } from '@aws-amplify/ui-react';
 
-import { DemoBox } from './DemoBox';
-import { FieldLabeler } from './FieldLabeler';
+import { DemoBox } from '@/components/DemoBox';
```
Comment (zchenwei) [inline docs/src/pages/components/table/TablePropControls.tsx:10]: We need to get rid of `DemoBox` in favor of `Demo` and show code generation

**Review by zchenwei — CHANGES_REQUESTED** [review]
Looking good. Left a few comments :)

File: docs/src/pages/components/table/TablePropControls.tsx [hunk 3]
```diff
@@ -32,7 +32,7 @@ export const TablePropControls: TablePropControlsInterface = ({
   size,
   variation,
 }) => (
-  <DemoBox primitiveName="Table">
+  <Flex direction="column">
```
[the remark this answers is not in the document]
Comment (jacoblogan) [reply docs/src/pages/components/table/TablePropControls.tsx:33]: Yes good catch, thanks!

**Review by zchenwei — APPROVED** [review]
LGTM! :shipit:

**Review by dbanksdesign — APPROVED** [review]

**Pull request merged**